services/src/main/java/org/keycloak/vault/DefaultVaultRawSecret.java

```diff
@@ -39,11 +39,15 @@ public void close() {
 
     private final ByteBuffer rawSecret;
 
-    public static VaultRawSecret forBuffer(Optional<ByteBuffer> buffer) {
-        if (buffer == null || ! buffer.isPresent()) {
+    public static VaultRawSecret empty() {
+        return EMPTY_VAULT_SECRET;
+    }
+
+    public static VaultRawSecret forByteArray(byte[] arr) {
+        if (arr == null || arr.length == 0) {
```

Comment:
The `arr.length` should be removed to distinguish between existing but empty secret and nonexistent one (`null`)

Reply:
Let me resolve this comment and focus on a discussion in the `PlainTextVaultProvider.java `. It will be clearer from that perspective.